examples/applications/author-disambiguation/sampling.py

```diff
@@ -0,0 +1,225 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Beard.
+# Copyright (C) 2015 CERN.
+#
+# Beard is a free software; you can redistribute it and/or modify it
+# under the terms of the Revised BSD License; see LICENSE file for
+# more details.
+
+"""
+.. codeauthor:: Hussein Al-Natsheh <hussein.al.natsheh@cern.ch>
+.. codeauthor:: Mateusz Susik <mateusz.susik@cern.ch>
+
+"""
+
+from __future__ import print_function
+
+import argparse
+import json
+import numpy as np
+import random
+import pickle
+
+from sklearn.cross_validation import train_test_split
+
+from beard.utils import normalize_name
+from beard.clustering import block_double_metaphone
+from beard.clustering import block_last_name_first_initial
+
+
+def unique_rows(list):
+
+    list = np.array(list)
+    x = np.dtype((np.void, list.dtype.itemsize * list.shape[1]))
+    b = np.ascontiguousarray(list).view(x)
+    _, idx = np.unique(b, return_index=True)
+
+    return list[idx]
+
+
+def sampling_pairs(blocking_function,
+                   blocking_threshold,
+                   sample_size,
+                   signatures_filename,
+                   clusters_filename,
+                   test_size, random_state,
+                   balanced, verbose):
+    """Sampling pairs from the ground-truth data.
+
+        Parameters
+        ----------
+        :param blocking_function: string
+            must be a defined blocking function. Defined functions are:
+            - block_last_name_first_initial
+            - block_double_metaphone
+
+        :param blocking_threshold: int or None
+            It determines the maximum allowed size of bloking on the last name.
+            It can only be:
+            -   None; if the blocking function is block_last_name_first_initial
+            -   int; if the blocking function is block_double_metaphone
+
+        :param sample_size: integer
+            The desired sample size
+
+        :param signatures_filename: string
+            The path where the input signitures file
+
+        :param clusters_filename: string
+            The path where the input clusters (ground-truth) file
+
+        :param test_size: float
+            The size of the test set: 0.0 < size < 1.0.
+            It is used for train_test_split. Should be the same as in clustering
+
+        :param random_state: integer
+            a randome_state value.
```

Comment:
In fact, I would change the API and make this function accepts a list of signatures, where these signatures are assumed to be already coming from the training set. This way, train/test split would happen only at one place and this function would focus on what it should: drawing pairs.